common/src/main/java/org/apache/drill/common/exceptions/UserException.java

```diff
@@ -536,6 +542,33 @@ public Builder pushContext(final String name, final double value) {
      * @return user exception
      */
     public UserException build(final Logger logger) {
+
+      // To allow for debugging:
+      //     A spinner code to make the execution stop here while the file '/tmp/drillspin' exists
+      // Can be used to attach a debugger, use jstack, etc
+      // The processID of the spinning thread should be in a file like /tmp/spin4148663301172491613.tmp
+      // along with the error message.
+      File spinFile = new File("/tmp/drillspin");
```

Comment:
Should this be a config setting? Probably the config is not visible here, but can we set a static variable at start-up time? And, since this code will check the file system on every exception, should we have a config variable to turn on the check?

Feel free to tell me I'm being overly paranoid...

Reply:
Using a "flag file" instead of a config setting gives more flexibility; like no need to restart in order to turn this feature on/off, or can select to catch errors only in few nodes, and last -- can free the looping thread by deleting this "flag file". 
      I also plan on posting an announcement on the dev list about this new "feature", and see if there's any feedback.